flang/lib/Optimizer/Builder/FIRBuilder.cpp

```diff
@@ -271,27 +271,29 @@ mlir::Value fir::FirOpBuilder::createHeapTemporary(
 
 /// Create a global variable in the (read-only) data section. A global variable
 /// must have a unique name to identify and reference it.
-fir::GlobalOp fir::FirOpBuilder::createGlobal(mlir::Location loc,
-                                              mlir::Type type,
-                                              llvm::StringRef name,
-                                              mlir::StringAttr linkage,
-                                              mlir::Attribute value,
-                                              bool isConst, bool isTarget) {
+fir::GlobalOp fir::FirOpBuilder::createGlobal(
+    mlir::Location loc, mlir::Type type, llvm::StringRef name,
+    mlir::StringAttr linkage, mlir::Attribute value, bool isConst,
+    bool isTarget, fir::CUDAAttributeAttr cudaAttr) {
   auto module = getModule();
   auto insertPt = saveInsertionPoint();
   if (auto glob = module.lookupSymbol<fir::GlobalOp>(name))
     return glob;
   setInsertionPoint(module.getBody(), module.getBody()->end());
-  auto glob =
-      create<fir::GlobalOp>(loc, name, isConst, isTarget, type, value, linkage);
+  llvm::SmallVector<mlir::NamedAttribute> attrs;
+  if (cudaAttr)
+    attrs.push_back(mlir::NamedAttribute(
+        mlir::StringAttr::get(module.getContext(), "cuda_attr"), cudaAttr));
```

Comment:
Could you please use the generated attribute name getter instead of the literal "cuda_attr"?

Reply:
Leftover from early test. Sure I'll update it.

Reply:
I updated it but unfortunately we cannot use the generated getter without an object. So I added a getter in the extra declaration part. This is has been an issue for long time and we should maybe try to fix it in MLIR tablegen at some point. This is the reason we have constexpr getter all over the extra declaration of our FIR operations.

Reply:
Thank you, Valentin!

We could use something like this: https://github.com/llvm/llvm-project/pull/79301/files#diff-aa56bf8cfc0c7b570e4e825e0a9e375a930085bc6837ca4854557d0c4c0ab977R58

I am okay with the way it is now.

Reply:
Yeah we can! I was looking for a way to get the OperationName object and looks like we can do it easily. I'll update the patch. I like to avoid duplication of this information